### ISSUE TITLE
Pagerduty integration fail when the alert rule description contains special characters

### ISSUE THREAD

**Issue by thomasgueze** [wazuh/wazuh#3135]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.6.1 | Manager | Packages | Centos 7 |

Hi,  

I'm using Wazuh with the PagerDuty integration.  
It works fine except when the integration is called for an alert mapped to a rule that contains a ' character in the description.  

Example :

~~~txt
<rule id="<id>" level="<level>">
  ... some rules criterias ...
  <description>Some description with a ' character</description>
</rule>
~~~

The pagerduty script is called but integratord fail.

From what I have been debugging, it's when it reaches in the pagerduty shell script :

https://github.com/wazuh/wazuh/blob/d1528e5577c0b0289d5285f1f1b9e7e3781342e9/integrations/pagerduty#L43

The script generated by integratord does not escape special characters :

https://github.com/wazuh/wazuh/blob/d1528e5577c0b0289d5285f1f1b9e7e3781342e9/src/os_integrator/integrator.c#L370-L374

When the script is called, this call does not succeed (and hang ?).  
So the daemon fail and I haven't checked too far after that but it seems that no more integrations are triggered after that (a restart of ossec or integratord seem to resolve this issue), I guess integratord is waiting for the script to finish.  

I have encountered this issue with the ' character but I imagine it may fail for others special characters.

**Comment by vikman90** [comment]
Hi @thomasgueze,

I'm trying to reproduce this issue, but I have the same result whether the rule description includes a `'` or not. Let us work on it.

In addition, I've seen that this integration always fails on Debian 9 because `/bin/sh` links to `/bin/dash` instead of `/bin/bash` and that produces a syntax error.

Thank you for reporting this bug. I will keep you posted.
Best regards.

**Comment by thomasgueze** [comment]
Thanks.

If you need additional information, happy to help.

**Comment by vikman90** [comment]
@thomasgueze Apologies for the late reply.

I've given it another try, I managed to reproduce your issue with this rule:
```xml
<rule id="100002" level="5">
  <match>testing</match>
  <description>Testing rule'</description>
</rule>
```

Then the script printed this error:
> /var/ossec/integrations/pagerduty: 7: /tmp/pagerduty-1560174224--770921932.alert: Syntax error: Unterminated quoted string

Fortunately, this issue has been fixed by @spartantri (#2403) and applied in Wazuh 3.9.0.

Upgrading your manager to the latest version of Wazuh will fix the problem. If this is not possible, maybe you want to update the PagerDuty script only.

Let me suggest this script to upgrade the PagerDuty integration:
```sh
#! /bin/bash
[ "$(whoami)" = "root" ] || exit 1
. /etc/ossec-init.conf || exit 1
PAGERDUTY="$DIRECTORY/integrations/pagerduty"
curl -Lo $PAGERDUTY https://raw.githubusercontent.com/wazuh/wazuh/v3.9.0/integrations/pagerduty
chown root:ossec $PAGERDUTY
chmod 750 $PAGERDUTY
```

Hope it helps.
Best regards.

**Comment by thomasgueze** [comment]
Great, thanks for the analysis and response !

I will check out the update, this should indeed resolve it.
